### ISSUE TITLE
Possible issue with test files?

### ISSUE THREAD

**Issue by Marie-Burns22** [learn-co-curriculum/react-event-system-lab#9]
Hi, my code works in the browser and pasts all the test when `npm test` is run. However, running `learn` produces errors all related to test files and does not trigger the "Run Local Tests" icon in the learn lab page to register that the tests have run or passed.  Running `learn submit` seems to work correctly.  I've pasted in the errors from running `learn` below.
`// ♥ learn
/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha/lib/reporters/base.js:250
  runner.stats = stats;
               ^

TypeError: Cannot set property stats of #<EventEmitter> which has only a getter
    at Spec.Base (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha/lib/reporters/base.js:250:16)
    at new Spec (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha/lib/reporters/spec.js:24:8)
    at withReplacedStdout (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:220:21)
    at withReplacedStdout (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:146:12)
    at setup.map (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:216:14)
    at Array.map (<anonymous>)
    at initReportersAndStreams (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:208:6)
    at mochaMulti (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:267:31)
    at new MochaMulti (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha-multi/mocha-multi.js:293:25)
    at Mocha.run (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha/lib/mocha.js:520:18)
    at Object.<anonymous> (/Users/marieburns/Development/code/react-event-system-lab-v-000/node_modules/mocha/bin/_mocha:480:18)
    at Module._compile (internal/modules/cjs/loader.js:738:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:749:10)
    at Module.load (internal/modules/cjs/loader.js:630:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:570:12)
    at Function.Module._load (internal/modules/cjs/loader.js:562:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:801:12)
    at internal/main/run_main_module.js:21:11
[09:21:00] (master) react-event-system-lab-v-000
// ♥ Traceback (most recent call last):
        11: from /Users/marieburns/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
        10: from /Users/marieburns/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
         9: from /Users/marieburns/.rvm/gems/ruby-2.6.1/bin/learn-test:23:in `<main>'
         8: from /Users/marieburns/.rvm/gems/ruby-2.6.1/bin/learn-test:23:in `load'
         7: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/bin/learn-test:68:in `<top (required)>'
         6: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/runner.rb:20:in `run'
         5: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/runner.rb:20:in `fork'
         4: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/runner.rb:21:in `block in run'
         3: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/runner.rb:44:in `report_and_clean'
         2: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/reporter.rb:13:in `report'
         1: from /Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/reporter.rb:47:in `report'
/Users/marieburns/.rvm/gems/ruby-2.6.1/gems/learn-test-2.6.1/lib/learn_test/strategies/mocha.rb:42:in `results': undefined method `[]' for nil:NilClass (NoMethodError)
`

**Comment by Madeline-Stark** [comment]
This is still an issue, except npm test gave this error: sh: mocha: command not found

I was not able to find another way of running the tests.

**Comment by maxwellbenton** [comment]
Hi @Marie-Burns22,

Thank you for providing this feedback. Unfortunately, I was unable to replicate the exact error you experienced, but this issue did help identify other errors due to the test dependencies being outdated. We've updated the dependencies for this lab and some of the test configuration to align with other labs in this section and have confirmed that the tests are working as expected currently. I also tested the updates using the solution you submitted and confirmed that it works with the current test setup.

We apologize for any frustration this lab may have caused you, but we appreciate you taking the time to provide feedback and help us improve our curriculum.

Thank you!
